AutomatticTracks/src/main/java/com/automattic/android/tracks/crashlogging/CrashLoggingDataProvider.kt

```diff
@@ -50,4 +50,16 @@ interface CrashLoggingDataProvider {
      * of crash logging data collection
      */
     fun userHasOptOutProvider(): Boolean
+
+    /**
```

Comment:
💡 Idea: Have you considered providing a single method, something like `fun updateEventExtrasBeforeSend(extras: Map<String, Object>): Map<String, Object>`? The library would provide a map of key-value pairs and it'd be up to the consumer to set whatever they want. It feels a bit weird that the consumer needs to inform the library about possible keys first and then the library iterates over these keys and asks the client what values to set. Wdyt?

Reply:
I get your point @malinajirka . It's a good suggestion unfortunately current Sentry API doesn't allow that.

To make sure if I understood you correctly: by writing 
> The library would provide a map of key-value pairs

you mean `map of key-value pairs` of the `SentryEvent` right? So accessor to it is package-private. See: https://github.com/getsentry/sentry-java/blob/4.3.0/sentry/src/main/java/io/sentry/SentryEvent.java#L264

![image](https://user-images.githubusercontent.com/5845095/113994692-e2399a00-9855-11eb-8b8b-6e36360e062a.png)

I'll ask Sentry team about that, maybe they will consider making this property a `public` one but it will probably take some time, until the next Sentry release.

Reply:
I checked the method before writing the comment but I missed the missing "public" modifier 😞 . 

Nitpick: However, I'm still wondering if it isn't cleaner to support `fun getEventExtrasBeforeSend(): Map<String, Object>` and let the consumer return a map and then iterate over the map and set the properties to Sentry. Wdyt?

Reply:
> I'm still wondering if it isn't cleaner to support fun getEventExtrasBeforeSend(): Map<String, Object>

We have to know current extra values before preparing new one. E.g. `encrypted logs` feature will send encrypted logs and then receive `UUID` for those encrypted logs identification. If we won't check for existing `encrypted logs` `UUID` in `SentryEvent`, we would end up sending those logs more than once (this case is described [here](https://github.com/wordpress-mobile/WordPress-Android/blob/a9eeed79f6d85fd3a3a55157ff9b2932248c00cc/WordPress/src/main/java/org/wordpress/android/util/CrashLogging.kt#L57-L66)).

Your comment inspired me though to pass all `known keys to value` map in signature of a method that prepares extras to `SentryEvent`. This way e.g. we can decide in client if we want to assing a new value for existing key, or if we want to just pass it further. https://github.com/Automattic/Automattic-Tracks-Android/pull/80/commits/878def0cfcf489b35ff15622ba6779018e4e16f2

I'll re-request review as this is quite different from initially proposed. Let me know what you think!

Reply:
Thanks for the clarification and improvements ;)!